kafka-ui-react-app/src/components/common/table/TableHeaderCell/TableHeaderCell.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { TopicColumnsToSort } from 'generated-sources';
+import StyledTableHeaderCell from 'components/common/table/TableHeaderCell/TableHeaderCell.styled';
+import cx from 'classnames';
+
+export interface TableHeaderCellProps {
+  title?: string;
+  thType?: 'primary';
+  previewText?: string;
+  onPreview?: () => void;
+  orderBy?: TopicColumnsToSort | null;
+  orderValue?: TopicColumnsToSort | null;
+  onOrderBy?: (orderBy: TopicColumnsToSort | null) => void;
+}
+
+const TableHeaderCell: React.FC<TableHeaderCellProps> = (props) => {
+  const { title, previewText, onPreview, orderBy, orderValue, onOrderBy } =
+    props;
+
+  return (
+    <StyledTableHeaderCell
+      className={cx(orderBy && orderBy === orderValue && 'has-text-link-dark')}
+      {...props}
+    >
+      <span className="title">{title}</span>
+      {previewText && (
+        <span
+          className="preview"
+          onClick={onPreview}
+          onKeyDown={onPreview}
+          role="button"
+          tabIndex={0}
+        >
+          {previewText}
+        </span>
+      )}
+      {orderValue && (
+        <span
+          className="icon is-small is-clickable"
+          onClick={() => orderValue && onOrderBy && onOrderBy(orderValue)}
```

Comment:
I think the whole header should be clickable

Reply:
in figma there is design for table header cell with "preview" button (I assume it is button) so I separate "preview",  "orderBy" and th element